### DISCUSSION TITLE
UGS connects fine, runs spindle, but no stepper motion…

### DISCUSSION THREAD

**Discussion by DarylBurton** [winder/Universal-G-Code-Sender#2626] [Help] [answered]
The UGS installation on my Mac (Catalina 10.15.7) went seamlessly, and connects to my generic 3018 machine (fitted with an Annoy Tools GRBL1.1 board) remarkably quickly and without fuss.

The console reports successful connection ok: Grbl 1.1f 

The power switch on the 24v supply is on (with a light showing) and the power supply is showing 24.5v on a multimeter

There’s a flashing green TX LED on the board.

The green RX LED on the board also flashes when you use the jog settings in UGS.

The spindle can be made to run by manually entering M3S1000 into the console.

But, I get absolutely no reaction from any of the (Cronos) stepper motors when jogging, or when using the setup wizard to check the correct direction of each axis. The stepper motors murmured just once during the setup process, but haven’t moved in a meaningful manner. Additionally. It’s possible to manually turn the stepper motors with the board connected and receiving jogging instructions, no resistance at all.  

Recollections from a previous project using stepper motors is that they do quietly let you know they’re there, even when they’re not being asked to do anything useful!

Is there anything obvious here in the software setup that I’ve omitted? I‘ve treble checked all the board to motor connections and it seems unlikely that three stepper motors are all DOA. 

Anybody come across this / have any thoughts or a fix / suggestions of anything else I can test?

Eg, does it matter in what order everything is powered up / connected together…?

Thanks!

**Comment by breiler** [comment]
Hi Daryl!

> But, I get absolutely no reaction from any of the (Cronos) stepper motors when jogging, or when using the setup wizard to check the correct direction of each axis. The stepper motors murmured just once during the setup process, but haven’t moved in a meaningful manner. 

This depends on the setting, can you post your settings `$$`.

> Is there anything obvious here in the software setup that I’ve omitted? I‘ve treble checked all the board to motor connections and it seems unlikely that three stepper motors are all DOA.

Can you upload a picture of your controller showing the wiring without the fan.

> Eg, does it matter in what order everything is powered up / connected together…?

The only problem I can think of is when connecting/disconnecting the stepper motor wires, which should be done when there is no power on. For instance, if you disconnect a stepper motor wire lead when the power is on, the inductance still present in the other wire can damage the driver circuit.

**Comment by DarylBurton** [comment]
Hi Joacim

Thanks so much for your reply, it is appreciated.

Here’s the settings and a couple of pictures:



$0 = 10    (Step pulse time, microseconds)
$1 = 25    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 6    (Step direction invert, mask)
$4 = 0    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 1    (Status report options, mask)
$11 = 1.000    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 0    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 7    (Homing direction invert, mask)
$24 = 25.000    (Homing locate feed rate, mm/min)
$25 = 500.000    (Homing search seek rate, mm/min)
$26 = 250    (Homing switch debounce delay, milliseconds)
$27 = 2.000    (Homing switch pull-off distance, millimeters)
$30 = 1000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$100 = 800.000    (X-axis travel resolution, step/mm)
$101 = 800.000    (Y-axis travel resolution, step/mm)
$102 = 800.000    (Z-axis travel resolution, step/mm)
$110 = 2000.000    (X-axis maximum rate, mm/min)
$111 = 2000.000    (Y-axis maximum rate, mm/min)
$112 = 100.000    (Z-axis maximum rate, mm/min)
$120 = 20.000    (X-axis acceleration, mm/sec^2)
$121 = 20.000    (Y-axis acceleration, mm/sec^2)
$122 = 20.000    (Z-axis acceleration, mm/sec^2)
$130 = 500.000    (X-axis maximum travel, millimeters)
$131 = 500.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
ok

![E3E5BF23-AB60-4FA0-B967-CDCB045EA30B_1_201_a](https://github.com/user-attachments/assets/2468a071-7733-46f1-8200-04fa4f650a1b)
![3B85CA79-F498-4332-8436-3E2D95E0DA75_1_201_a](https://github.com/user-attachments/assets/012bf7aa-4cdf-4e5d-810c-eef109992bf8)


Additionally, you’ll see that the x axis 4988 stepper driver is missing…. Full disclosure, a lot of what I was reading / images of other machines with very similar setups showed the green and blue stepper motor wires reversed from the position of mine, so I threw caution to the wind and swapped those two on one of the stepper motor cables. 

Followed, immediately when powering up by a tiny flash and puff of smoke from the stepper driver. The rest of the board seems unaffected and powers up / connects OK. The second picture shows the stepper motor connecter wired as supplied in the kit (not the one I fiddled with!)

I now feel even dumber than before and hope it hasn’t complicated any fault analysis… :-( 

I have also just ordered a couple of identical stepper drivers from a local eBay source!

All the best







Daryl

**Reply by breiler** [reply]
To make the motors engage all the time, change the setting `$1=255` (https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration#1---step-idle-delay-milliseconds)

>  Full disclosure, a lot of what I was reading / images of other machines with very similar setups showed the green and blue stepper motor wires reversed from the position of mine, so I threw caution to the wind and swapped those two on one of the stepper motor cables.

I think that it is strange that it would fry the driver. If you had it powered on when unplugging the stepper motor could explain it. The color scheme of the stepper motor cables may differ a bit between manufacturers. 

You can check and see if the controller still output a 5v logic output to these pins with a volt meter: EN, STEP and DIR
![image](https://github.com/user-attachments/assets/73ed8722-62ca-491d-8517-21714cd2676b)

On your board it should be located here:
![image](https://github.com/user-attachments/assets/737b34d0-7678-4e3b-9520-d1daf54f9895)


On the other channels for Y and Z you can try and decrease the the step/mm setting. Begin with `$101=80` and `$102=80` and see if you get any movement.

**Comment by DarylBurton** [comment]
Hi Joacim

Thanks again for your reply, and for making your guidance so straightforward to test for a non-tech! 

I’m afraid that engaging the motors all the time via $1=255 and decreasing the step/mm setting on all three axes to 80/mm $101=80, $102=80, $103=80 hasn’t resulted in any movement from the steppers via jog. 

Connection to the Mac via UGS is still pretty much instant with the TX LED showing (and RX LED showing on a jog command) and the power LEDs also light up on connection of external 24v power. 

The spindle can also still be made to operate via the console.

I did notice a very light humming from the machine on one occasion when jogging / checking the axes direction, but literally just a slight hum, and one that’s not reliably repeatable.

Any thoughts on what else I could be checking setup- or hardware-wise?

On that separate note to check if there was more on the X axis (in addition to the driver) that was damaged in the cable swap, I did test as you suggested and got 4.66v on DIR but 0v on STEP and ENABLE.

**Comment by DarylBurton** [comment]
![tempImageWkIVLc](https://github.com/user-attachments/assets/2b482059-a01c-471e-bca9-ed4473fbdb2b)
Some extra info, just in case it creates a lightbulb moment for anyone!

I downloaded UGS onto my partner’s Windows 10 laptop. This too connects almost instantaneously with the TX light on the board flashing, together with a flashing RX light when a jog command is sent.

Downloaded Candle onto the same laptop. This connects to the board too, with the same lights showing on connection / jog commands.

After changing the settings as Joacim suggested ($1=255, $101=80, $102=80, $103=80) still no stepper motor movement. 

So, I did start to wonder about four possibilities (from a position of immense ignorance, sorry!) 

1/ The stepper drivers have pots to adjust the current the stepper motors receive. I’m guessing it’s possible to have that reduced so far that the motors aren’t able to move at all? However, I couldn’t interpret the code on the adjacent sensing resistor to calculate what Vref should be, so thought I was probably already out of my depth (picture attached in case you can make sense of the code. Looks like 303 to me, but this doesn’t tie up with any small resistor I can find!).

2/ Although the board seems to connect, maybe the board itself is faulty in some other way, providing no output to the stepper motors regardless of its apparent connection state to UGS?

3/ Ignoring the previous slight excitement when a couple of the stepper motor wires were previously swapped over, maybe the stepper motor wiring is incorrect after all? I do understand that the actual colour of the wires is immaterial, but I’ve attached a photo so it’s possible to see from which pin out on the driver board (four in total) goes to which pin in on the stepper motor (six available). Looking from left to right it’s presently set up as: 

Driver black wire (pin 1) out: to pin 1 in on the motor.

Driver green wire (pin 2) out: to pin 4 in on the motor.

Driver blue wire (pin 3) out: to pin 3 in on the motor.

Driver red wire (pin 4) out: to pin 6 in on the motor.

4/ maybe I should just order another generic 3018 board with leads and connect that up instead!

Any thoughts / advice would be very welcome :-)

![E0FB337D-CD24-4F84-899E-36E270D398B1_1_201_a](https://github.com/user-attachments/assets/3389147c-eafe-4a09-9d1f-75b606db563c)

**Reply by DarylBurton** [reply, marked as answer]
Just in case it helps anyone else with similar issues, replacing all the A4988 stepper driver / stepsticks got things working immediately.

Difficult to believe that all three could be DOA, but I’m guessing that I either did something dumb early on which rendered them all unserviceable, or maybe I was unlucky enough to get a board with all the drivers from the same bad manufacturing batch.

Might be worth swapping yours out if you have similar problems.

Incidentally, the UGS is working beautifully; installing quickly and simply on both Mac and Windows 10, connecting instantly to the Cronos / Annoy board, and now driving the 3018 machine very smoothly with a simple 2D file made in Fusion and CAMd through Estlcam. 

Really grateful to the developers. Thank you!
